lms/djangoapps/teams/csv.py

```diff
@@ -368,8 +370,25 @@ def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
                 'Team {} cannot have Master’s track users mixed with users in other tracks.'.format(team_name)
             self.add_error_and_check_if_max_exceeded(error_message)
             return False
+        if not self.is_enrollment_protection_for_existing_team_matches_user(user, team_name, teamset_id):
+            error_message = \
+                'User {} does not have access to team {}.'.format(user.username, team_name)
+            self.add_error_and_check_if_max_exceeded(error_message)
+            return False
         return True
 
+    def is_enrollment_protection_for_existing_team_matches_user(self, user, team_name, teamset_id):
```

Comment:
maybe `does_team_protection_status_match_user` (or visa versa?)

Reply:
I prefer to keep is as it matches existing boolean pattern.